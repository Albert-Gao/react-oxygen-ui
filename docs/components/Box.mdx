---
name: Box
menu: Components
---

import { Playground, PropsTable } from 'docz';
import { Box } from '../../src';

# Box

`<Box>` is probably the most popular component in the library. It's a flexbox actually. Just use it when you want a `div`. But it has some handy props to make your life easier.

## Properties

All `props` from `<BaseBox>`.

### gap

> gap: Responsive

It's the gap between each children inside this `Box`. From now on, you don't need to add margin to each children. You just add this one `gap` prop in the parent `Box`.

```javascript
<Box gap="10px">
  <Box>1</Box>
  <Box>2</Box>
  <Box>3</Box>
</Box>
```

### fill

> fill: 'horinzontal' | 'vertical'

- horinzontal: `width: 100%`
- vertical: `height: 100%`

<Playground>
  <Box fill="horinzontal" bg="yellow">
    Box<span>1</span>
  </Box>
</Playground>
