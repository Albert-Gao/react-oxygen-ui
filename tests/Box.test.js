import 'jest-styled-components';
import 'jest-dom/extend-expect';
import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { Box } from '../src/components/Box/Box';
import { css as cssFunc } from 'styled-components';
import { defaultTheme } from '../src/styles/IThemeType';
import { TestThemeProvider as ThemeProvider } from './utils';

const getMediaQuery = num =>
  `screen and (min-width:${typeof num === 'string' ? num : num + 'px'})`;

afterEach(cleanup);

describe('Test <Box />: justify-content', () => {
  test('should convert start to flex-start', () => {
    const { firstChild } = render(<Box justifyContent="start" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('justify-content', 'flex-start');
  });

  test('should convert end to flex-end', () => {
    const { firstChild } = render(<Box justifyContent="end" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('justify-content', 'flex-end');
  });

  test('should convert between to space-between', () => {
    const { firstChild } = render(<Box justifyContent="between" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('justify-content', 'space-between');
  });

  test('should convert around to space-around', () => {
    const { firstChild } = render(<Box justifyContent="around" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('justify-content', 'space-around');
  });
});

describe('Test <Box />: align-items', () => {
  test('should convert start to flex-start', () => {
    const { firstChild } = render(<Box alignItems="start" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('align-items', 'flex-start');
  });

  test('should convert end to flex-end', () => {
    const { firstChild } = render(<Box alignItems="end" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('align-items', 'flex-end');
  });
});

describe('Test <Box />: align-self', () => {
  test('should convert start to flex-start', () => {
    const { firstChild } = render(<Box alignSelf="start" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('align-self', 'flex-start');
  });

  test('should convert end to flex-end', () => {
    const { firstChild } = render(<Box alignSelf="end" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('align-self', 'flex-end');
  });
});

describe('Test <Box />: flex-direction', () => {
  test('should accept row', () => {
    const { firstChild } = render(<Box direction="row" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('flex-direction', 'row');
  });

  test('should accept column', () => {
    const { firstChild } = render(<Box direction="column" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('flex-direction', 'column');
  });
});

describe('Test <Box />: fill', () => {
  test('should support fill="horizontal"', () => {
    const { firstChild } = render(<Box fill="horizontal" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).not.toHaveStyleRule('height', '100%');
    expect(firstChild).toHaveStyleRule('width', '100%');
  });

  test('should support fill="vertical"', () => {
    const { firstChild } = render(<Box fill="vertical" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('height', '100%');
    expect(firstChild).not.toHaveStyleRule('width', '100%');
  });

  test('should support fill=true', () => {
    const { firstChild } = render(<Box fill />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('height', '100%');
    expect(firstChild).toHaveStyleRule('width', '100%');
  });

  test('should support fill=false', () => {
    const { firstChild } = render(<Box fill={false} />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).not.toHaveStyleRule('height', '100%');
    expect(firstChild).not.toHaveStyleRule('width', '100%');
  });
});

describe('Test <Box /> gap', () => {
  test('[no gap with children] it should not add gap between children', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box direction="row">
          <p>a</p>
          <p>b</p>
        </Box>
      </ThemeProvider>,
    ).container;
    // firstChild is <Box>
    const a = firstChild.children[0];
    const b = firstChild.children[1];

    expect(firstChild).toMatchSnapshot();

    // a + b = 2
    expect(firstChild.children.length).toEqual(2);
    expect(a.innerHTML).toEqual('a');
    expect(b.innerHTML).toEqual('b');
  });

  test('[direction=row] it should add gap between children', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box direction="row" gap="32px">
          <p>a</p>
          <p>b</p>
        </Box>
      </ThemeProvider>,
    ).container;
    // firstChild is <Box>
    const a = firstChild.children[0];
    const gapA = firstChild.children[1];
    const b = firstChild.children[2];

    expect(firstChild).toMatchSnapshot();

    // a + gapA + b = 3
    expect(firstChild.children.length).toEqual(3);
    expect(a.innerHTML).toEqual('a');
    expect(gapA.tagName).toEqual('DIV');
    expect(gapA).toHaveStyleRule('width', `32px`);
    expect(b.innerHTML).toEqual('b');
  });

  test('[direction=undefined] it should add gap between children', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box gap="12px">
          <p>a</p>
          <p>b</p>
        </Box>
      </ThemeProvider>,
    ).container;
    // firstChild is <Box>
    const a = firstChild.children[0];
    const gapA = firstChild.children[1];
    const b = firstChild.children[2];

    expect(firstChild).toMatchSnapshot();

    // a + gapA + b = 3
    expect(firstChild.children.length).toEqual(3);
    expect(a.innerHTML).toEqual('a');
    expect(gapA.tagName).toEqual('DIV');
    expect(gapA).toHaveStyleRule('width', `12px`);
    expect(b.innerHTML).toEqual('b');
  });

  test('it should add gap without ThemeProvider', () => {
    const { firstChild } = render(
      <Box gap="12px">
        <p>a</p>
        <p>b</p>
      </Box>,
    ).container;
    // firstChild is <Box>
    const a = firstChild.children[0];
    const gapA = firstChild.children[1];
    const b = firstChild.children[2];

    expect(firstChild).toMatchSnapshot();

    // a + gapA + b = 3
    expect(firstChild.children.length).toEqual(3);
    expect(a.innerHTML).toEqual('a');
    expect(gapA.tagName).toEqual('DIV');
    expect(gapA).toHaveStyleRule('width', `12px`);
    expect(b.innerHTML).toEqual('b');
  });

  test('[direction=column] it should add gap between children', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box direction="column" gap="33px">
          <p>a</p>
          <p>b</p>
        </Box>
      </ThemeProvider>,
    ).container;
    // firstChild is <Box>
    const a = firstChild.children[0];
    const gapA = firstChild.children[1];
    const b = firstChild.children[2];

    expect(firstChild).toMatchSnapshot();

    // a + gapA + b = 3
    expect(firstChild.children.length).toEqual(3);
    expect(a.innerHTML).toEqual('a');
    expect(gapA.tagName).toEqual('DIV');
    expect(gapA).toHaveStyleRule('height', `33px`);
    expect(b.innerHTML).toEqual('b');
  });

  test('[Multiple children] it should add gap between children', () => {
    const expectedLength = `55px`;
    const { firstChild } = render(
      <ThemeProvider>
        <Box direction={['row', 'column', 'row']} gap="55px">
          <p>a</p>
          <p>b</p>
          <p>c</p>
        </Box>
      </ThemeProvider>,
    ).container;
    // firstChild is <Box>
    const a = firstChild.children[0];
    const gapA = firstChild.children[1];
    const b = firstChild.children[2];
    const gapB = firstChild.children[3];
    const c = firstChild.children[4];

    expect(firstChild).toMatchSnapshot();

    // a + gapA + b + gapB + c = 5
    expect(firstChild.children.length).toEqual(5);
    expect(a.innerHTML).toEqual('a');
    expect(gapA.tagName).toEqual('DIV');
    expect(gapA).toHaveStyleRule('width', expectedLength);
    expect(gapA).toHaveStyleRule('width', expectedLength, {
      media: getMediaQuery(defaultTheme.breakpoints[0]),
    });
    expect(gapA).toHaveStyleRule('height', expectedLength, {
      media: getMediaQuery(defaultTheme.breakpoints[1]),
    });
    expect(b.innerHTML).toEqual('b');
    expect(gapB.tagName).toEqual('DIV');
    expect(gapB).toHaveStyleRule('width', expectedLength);
    expect(gapB).toHaveStyleRule('width', expectedLength, {
      media: getMediaQuery(defaultTheme.breakpoints[0]),
    });
    expect(gapB).toHaveStyleRule('height', expectedLength, {
      media: getMediaQuery(defaultTheme.breakpoints[1]),
    });
    expect(c.innerHTML).toEqual('c');
  });
});

describe('Test <Box />: multiple props', () => {
  test('should accept boxShadow', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box boxShadow={0} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('box-shadow', defaultTheme.shadows[0]);
  });

  test('should accept fontSize', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box fontSize="11px" />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('font-size', '11px');
  });

  test('should accept mx', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box mx="111px" />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('margin-left', '111px');
    expect(firstChild).toHaveStyleRule('margin-right', '111px');
  });

  test('should accept py', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box py="3px" />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('padding-top', '3px');
    expect(firstChild).toHaveStyleRule('padding-bottom', '3px');
  });

  test('should accept px[]', () => {
    const getSpace = key => defaultTheme.space[key] + 'px';
    const { firstChild } = render(
      <ThemeProvider>
        <Box px={[2, 3, 4]} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();

    expect(firstChild).toHaveStyleRule('padding-left', getSpace(3), {
      media: getMediaQuery(defaultTheme.breakpoints[0]),
    });

    expect(firstChild).toHaveStyleRule('padding-right', getSpace(3), {
      media: getMediaQuery(defaultTheme.breakpoints[0]),
    });

    expect(firstChild).toHaveStyleRule('padding-left', getSpace(4), {
      media: getMediaQuery(defaultTheme.breakpoints[1]),
    });

    expect(firstChild).toHaveStyleRule('padding-right', getSpace(4), {
      media: getMediaQuery(defaultTheme.breakpoints[1]),
    });

    expect(firstChild).toHaveStyleRule(
      'padding-left',
      defaultTheme.space[2] + 'px',
    );
    expect(firstChild).toHaveStyleRule(
      'padding-right',
      defaultTheme.space[2] + 'px',
    );
  });

  test('should accept py[]', () => {
    const getSpace = key => defaultTheme.space[key] + 'px';
    const { firstChild } = render(
      <ThemeProvider>
        <Box py={[3, 4, 5]} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();

    expect(firstChild).toHaveStyleRule('padding-top', getSpace(4), {
      media: getMediaQuery(defaultTheme.breakpoints[0]),
    });

    expect(firstChild).toHaveStyleRule('padding-bottom', getSpace(4), {
      media: getMediaQuery(defaultTheme.breakpoints[0]),
    });

    expect(firstChild).toHaveStyleRule('padding-top', getSpace(5), {
      media: getMediaQuery(defaultTheme.breakpoints[1]),
    });

    expect(firstChild).toHaveStyleRule('padding-bottom', getSpace(5), {
      media: getMediaQuery(defaultTheme.breakpoints[1]),
    });

    expect(firstChild).toHaveStyleRule(
      'padding-top',
      defaultTheme.space[3] + 'px',
    );
    expect(firstChild).toHaveStyleRule(
      'padding-bottom',
      defaultTheme.space[3] + 'px',
    );
  });

  test('should accept a11yTitle', () => {
    const a11attr = 'aria-label';
    const a11text = 'accessibility test';

    const { firstChild } = render(
      <ThemeProvider>
        <Box a11yTitle={a11text} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveAttribute(a11attr);
    expect(firstChild.getAttribute(a11attr)).toEqual(a11text);
  });
});

describe('Test <Box css={}/>: ', () => {
  test('css prop should be added', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box css={'background-color: black;'} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('background-color', 'black');
  });

  test('css prop (object) should be handled', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box css={{ backgroundColor: 'black', paddingLeft: '1023px' }} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('background-color', 'black');
    expect(firstChild).toHaveStyleRule('padding-left', '1023px');
  });

  test('css prop (css string) should be handled', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box
          css={cssFunc`
          background-color: black;
          padding-left: 1023px;
          `}
        />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('background-color', 'black');
    expect(firstChild).toHaveStyleRule('padding-left', '1023px');
  });
});
