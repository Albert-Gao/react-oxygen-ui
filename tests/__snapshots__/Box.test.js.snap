// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Box /> gap [Multiple children] it should add gap between children 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 55px;
}

@media screen and (min-width:40em) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    width: 55px;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    height: 55px;
  }
}

@media screen and (min-width:64em) {
  .c1 {
    width: 55px;
  }
}

<div
  class="c0"
  direction="row,column,row"
  display="flex"
>
  <p>
    a
  </p>
  <div
    class="c1"
    direction="row,column,row"
  />
  <p>
    b
  </p>
  <div
    class="c1"
    direction="row,column,row"
  />
  <p>
    c
  </p>
</div>
`;

exports[`Test <Box /> gap [direction=column] it should add gap between children 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 33px;
}

<div
  class="c0"
  direction="column"
  display="flex"
>
  <p>
    a
  </p>
  <div
    class="c1"
    direction="column"
  />
  <p>
    b
  </p>
</div>
`;

exports[`Test <Box /> gap [direction=row] it should add gap between children 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 32px;
}

<div
  class="c0"
  direction="row"
  display="flex"
>
  <p>
    a
  </p>
  <div
    class="c1"
    direction="row"
  />
  <p>
    b
  </p>
</div>
`;

exports[`Test <Box /> gap [direction=undefined] it should add gap between children 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 12px;
}

<div
  class="c0"
  display="flex"
>
  <p>
    a
  </p>
  <div
    class="c1"
    direction="row"
  />
  <p>
    b
  </p>
</div>
`;

exports[`Test <Box /> gap [no gap with children] it should not add gap between children 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  direction="row"
  display="flex"
>
  <p>
    a
  </p>
  <p>
    b
  </p>
</div>
`;

exports[`Test <Box /> gap it should add gap without ThemeProvider 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 12px;
}

<div
  class="c0"
  display="flex"
>
  <p>
    a
  </p>
  <div
    class="c1"
    direction="row"
  />
  <p>
    b
  </p>
</div>
`;

exports[`Test <Box />: align-items should convert end to flex-end 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: align-items should convert start to flex-start 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: align-self should convert end to flex-end 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: align-self should convert start to flex-start 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: fill should support fill="horizontal" 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
  width="100%"
/>
`;

exports[`Test <Box />: fill should support fill="vertical" 1`] = `
.c0 {
  box-sizing: border-box;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
  height="100%"
/>
`;

exports[`Test <Box />: fill should support fill=false 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: fill should support fill=true 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
  height="100%"
  width="100%"
/>
`;

exports[`Test <Box />: flex-direction should accept column 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  direction="column"
  display="flex"
/>
`;

exports[`Test <Box />: flex-direction should accept row 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  direction="row"
  display="flex"
/>
`;

exports[`Test <Box />: justify-content should convert around to space-around 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: justify-content should convert between to space-between 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: justify-content should convert end to flex-end 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: justify-content should convert start to flex-start 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: multiple props should accept a11yTitle 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  aria-label="accessibility test"
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: multiple props should accept boxShadow 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: 0px 1px 2px rgba(0,0,0,0.20);
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: multiple props should accept fontSize 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 11px;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
  font-size="11px"
/>
`;

exports[`Test <Box />: multiple props should accept mx 1`] = `
.c0 {
  box-sizing: border-box;
  margin-left: 111px;
  margin-right: 111px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: multiple props should accept px[] 1`] = `
.c0 {
  box-sizing: border-box;
  padding-left: 8px;
  padding-right: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

@media screen and (min-width:40em) {
  .c0 {
    padding-left: 16px;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    padding-left: 24px;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    padding-right: 16px;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    padding-right: 24px;
  }
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: multiple props should accept py 1`] = `
.c0 {
  box-sizing: border-box;
  padding-top: 3px;
  padding-bottom: 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: multiple props should accept py[] 1`] = `
.c0 {
  box-sizing: border-box;
  padding-top: 16px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

@media screen and (min-width:40em) {
  .c0 {
    padding-top: 24px;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    padding-top: 32px;
  }
}

@media screen and (min-width:40em) {
  .c0 {
    padding-bottom: 24px;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    padding-bottom: 32px;
  }
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box css={}/>:  css prop (css string) should be handled 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  background-color: black;
  padding-left: 1023px;
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box css={}/>:  css prop (object) should be handled 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  background-color: black;
  padding-left: 1023px;
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box css={}/>:  css prop should be added 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  background-color: black;
}

<div
  class="c0"
  display="flex"
/>
`;
